CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ if(${TIDY})
 endif()
 
 add_subdirectory(bl)
+add_subdirectory(gui)
 
 add_executable(tetris main/main.cpp)
-target_link_libraries(tetris PUBLIC blocks board)
+target_link_libraries(tetris PUBLIC blocks board frame_buffer)
```

Comment:
I'd say frame_buffer should be private dependency of the GUI lib (or what ever you want to call it), There is no need for the fianal user to even know there is such thing as frame buffer. The user should have only acces to drawing engine(interface and factory), abstract drawable and Color

Reply:
fixed